Backend/index.js

```diff
@@ -347,31 +383,60 @@ async function getTopSongs(accessToken) {
     }
 }
 
+async function saveTopSongs(username, songInfo){
+    const user = await findUser(username)
+    const latestTopSongs = await prisma.topSongs.findFirst({
+        where: {
+            userId: user.id
+        },
+        orderBy: {
+            createdAt: 'desc'
+        }
+    });
+
+    if (latestTopSongs && latestTopSongs.createdAt > Date.now() - 4 * 7 * 24 * 60 * 60 * 1000) { return latestTopSongs }
+    const topSongs = await prisma.topSongs.create({
+        data: {
+            tracks: {
+                create: songInfo.map((song) => ({
+                    track: song.songName,
+                    artist: song.artistNames.join(', ')
```

Comment:
not blocking - but is this the best way to represent artists? it may be harder to use for compatibility later on